From 655dbd1df912e14298ba26030d09a372bf03fade Mon Sep 17 00:00:00 2001
From: Oleksandr Tyshchenko <oleksandr_tyshchenko@epam.com>
Date: Thu, 22 Aug 2019 13:06:33 +0300
Subject: [PATCH 03/22] xen: Use large pages for MMIO mappings on ARM

To avoid using only 4K pages for establishing mappings for
large MMIO regions.

For example, the device specific region for Lossy Decompression (48M)
could be established using 2M pages...

Signed-off-by: Oleksandr Tyshchenko <oleksandr_tyshchenko@epam.com>
---
 xen/common/domctl.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/xen/common/domctl.c b/xen/common/domctl.c
index 21bbffdc40..5b4ae4c287 100644
--- a/xen/common/domctl.c
+++ b/xen/common/domctl.c
@@ -692,7 +692,7 @@ long do_domctl(XEN_GUEST_HANDLE_PARAM(xen_domctl_t) u_domctl)
 #ifndef CONFIG_X86 /* XXX ARM!? */
         ret = -E2BIG;
         /* Must break hypercall up as this could take a while. */
-        if ( nr_mfns > 64 )
+        if ( nr_mfns > 0x3000 )
             break;
 #endif
 
-- 
2.34.1

